vllm/platforms/__init__.py

```diff
@@ -1,20 +1,21 @@
-from typing import Optional
-
 import torch
 
-from vllm.utils import is_tpu
-
 from .interface import Platform, PlatformEnum, UnspecifiedPlatform
 
-current_platform: Optional[Platform]
+current_platform: Platform
+
+try:
+    import libtpu
+except ImportError:
+    libtpu = None
 
 if torch.version.cuda is not None:
     from .cuda import CudaPlatform
     current_platform = CudaPlatform()
 elif torch.version.hip is not None:
     from .rocm import RocmPlatform
     current_platform = RocmPlatform()
-elif is_tpu():
+elif libtpu is not None:
     from .tpu import TpuPlatform
     current_platform = TpuPlatform()
 else:
```

Comment:
While this implementation works for now, I think it's better to check `libtpu` before CUDA because PyTorch XLA might be used with PyTorch + CUDA (same for other hardware like gaudi).

Reply:
XLA and tpu are different concept, right? XLA can be used for several platforms, while `libtpu` is only used for tpu.

Reply:
We are checking TpuPlatform here, not XLAPlatform.

Reply:
I mean, when the installed pytorch version has CUDA, vLLM will recognized the underlying platform as NVIDIA GPU instead of TPUs.

Reply:
got it. will update.